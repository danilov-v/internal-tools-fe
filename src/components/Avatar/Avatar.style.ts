import styled from 'styled-components';

export const AvatarWrapper = styled.div`
  position: relative;
`;

export const AvatarImage = styled.img`
  width: 35px;
  height: 35px;
  display: inline-block;
  max-width: 100%;
`;

export const AvatarCaption = styled.div`
  display: inline-block;
  color: #cccccc;
  margin-left: 5px;

  span {
    position: absolute;
    bottom: 20px;
    font-size: 10px;
    color: #727374;
  }
`;
