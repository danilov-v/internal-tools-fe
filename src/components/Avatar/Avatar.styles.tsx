import styled from 'styled-components';

export const AvatarWrapper = styled.div`
  position: relative;
`;

export const AvatarImage = styled.img`
  width: 35px;
  height: 35px;
  display: inline-block;
  max-width: 100%;
`;

export const AvatarCaption = styled.div`
  display: inline-block;
  font-size: 20px;
  color: #cccccc;
  margin-left: 5px;

  span {
    position: absolute;
    bottom: 25px;
    font-size: 12px;
    color: #727374ff;
  }
`;
