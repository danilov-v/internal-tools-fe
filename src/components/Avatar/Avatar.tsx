import React, { ReactElement } from 'react';
import * as R from 'helpers/rank';
import captain from './captain_rang.svg';
import starley from './starley_rang.svg';
import leyt from './leyt_rang.svg';

import * as S from './Avatar.styles';

type AvatarProps = {
  user: {
    lastName: string;
    position: string;
    rank: string;
  };
};

const getRankImage = (rank: string): ReactElement => {
  switch (rank) {
    case R.CAPTAIN:
      return <S.AvatarImage src={captain} alt="Avatar" />;
    case R.SENIOR_LIEUTENANT:
      return <S.AvatarImage src={starley} alt="Avatar" />;
    case R.LIEUTENANT:
      return <S.AvatarImage src={leyt} alt="Avatar" />;
    default:
      return <S.AvatarImage src={leyt} alt="Avatar" />;
  }
};

export const Avatar: React.FC<AvatarProps> = ({ user }) => {
  return (
    <S.AvatarWrapper>
      {getRankImage(user.rank)}
      <S.AvatarCaption>
        <span>{user.position}</span>
        {user.lastName}
      </S.AvatarCaption>
    </S.AvatarWrapper>
  );
};
