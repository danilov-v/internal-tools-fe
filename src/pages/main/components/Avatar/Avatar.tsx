import React, { ReactElement } from 'react';

import * as R from 'helpers/rank';
import captain from 'assets/ranks/captain.svg';
import lieutenant from 'assets/ranks/lieutenant.svg';
import seniorLieutenant from 'assets/ranks/senior-lieutenant.svg';

import * as S from './Avatar.style';

type AvatarProps = {
  user: {
    lastName: string;
    position: string;
    rank: string;
  };
};

const getRankImage = (rank: string): ReactElement => {
  switch (rank) {
    case R.CAPTAIN:
      return <S.AvatarImage src={captain} alt="Avatar" />;
    case R.SENIOR_LIEUTENANT:
      return <S.AvatarImage src={seniorLieutenant} alt="Avatar" />;
    case R.LIEUTENANT:
      return <S.AvatarImage src={lieutenant} alt="Avatar" />;
    default:
      return <S.AvatarImage src={lieutenant} alt="Avatar" />;
  }
};

export const Avatar: React.FC<AvatarProps> = ({ user }) => {
  return (
    <S.AvatarWrapper>
      {getRankImage(user.rank)}
      <S.AvatarCaption>
        <span>{user.position}</span>
        {user.lastName}
      </S.AvatarCaption>
    </S.AvatarWrapper>
  );
};
