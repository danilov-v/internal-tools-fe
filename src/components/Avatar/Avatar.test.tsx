import React from 'react';
import { render } from '@testing-library/react';

import { Avatar } from './Avatar';

const testUser = {
  lastName: 'test',
  rank: 'captain',
  position: 'test position',
};

describe('<Avatar />', () => {
  it('should render and match snapshot', () => {
    const component = render(<Avatar user={testUser} />);

    expect(component.baseElement).toMatchSnapshot();
  });
});
