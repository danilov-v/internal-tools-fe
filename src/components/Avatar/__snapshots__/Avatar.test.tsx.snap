// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render and match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-AxjAm jfEBQj"
    >
      <img
        alt="Avatar"
        class="sc-AxirZ hmcxZZ"
        src="captain_rang.svg"
      />
      <div
        class="sc-AxiKw ffVKvN"
      >
        <span>
          test position
        </span>
        test
      </div>
    </div>
  </div>
</body>
`;
